oap-server/server-configuration/configuration-etcd/src/test/java/org/apache/skywalking/oap/server/configuration/etcd/ITEtcdConfigurationTest.java

```diff
@@ -0,0 +1,153 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.configuration.etcd;
+
+import java.io.FileNotFoundException;
+import java.io.Reader;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import mousio.etcd4j.EtcdClient;
+import mousio.etcd4j.promises.EtcdResponsePromise;
+import mousio.etcd4j.responses.EtcdKeysResponse;
+import org.apache.skywalking.apm.util.PropertyPlaceholderHelper;
+import org.apache.skywalking.oap.server.library.module.ApplicationConfiguration;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.util.CollectionUtils;
+import org.apache.skywalking.oap.server.library.util.ResourceUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Alan Lau
+ */
+public class ITEtcdConfigurationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ITEtcdConfigurationTest.class);
+
+    private final Yaml yaml = new Yaml();
+
+    private EtcdServerSettings settings;
+
+    private EtcdConfigurationTestProvider provider;
+
+    private EtcdClient client;
+
+    @Before
+    public void setUp() throws Exception {
+        final ApplicationConfiguration applicationConfiguration = new ApplicationConfiguration();
+        loadConfig(applicationConfiguration);
+
+        final ModuleManager moduleManager = new ModuleManager();
+        moduleManager.init(applicationConfiguration);
+
+        final String etcdHost = System.getProperty("etcd.host");
+        final String etcdPort = System.getProperty("etcd.port");
+        logger.info("etcdHost: {}, etcdPort: {}", etcdHost, etcdPort);
+        Properties properties = new Properties();
+        properties.setProperty("serverAddr", etcdHost + ":" + etcdPort);
+
+        List<URI> uris = EtcdUtils.parseProp(properties);
+        client = new EtcdClient(uris.toArray(new URI[] {}));
+
+        provider =
+            (EtcdConfigurationTestProvider)moduleManager
+                .find(EtcdConfigurationTestModule.NAME)
+                .provider();
+
+        assertNotNull(provider);
+    }
+
+    @Test(timeout = 2000000)
```

Comment:
`2000000 ms ~= 33 minutes`, does it take so long in reality? if not I'll suggest choosing a reasonable threshold, otherwise the build job will hang for a long time

Reply:
yes, i forget to change this. i am working on this.